Sources/FoundationNetworking/URLResponse.swift

```diff
@@ -46,9 +45,9 @@ open class URLResponse : NSObject, NSSecureCoding, NSCopying, @unchecked Sendabl
             self.textEncodingName = encodedEncodingName as String
         }
         
-        if let encodedFilename = aDecoder.decodeObject(of: NSString.self, forKey: "NS.suggestedFilename") {
-            self.suggestedFilename = encodedFilename as String
-        }
+        // re-sanitizing with lastPathComponent because of supportsSecureCoding
+        let encodedFilename = aDecoder.decodeObject(of: NSString.self, forKey: "NS.suggestedFilename")?.lastPathComponent
+        self.suggestedFilename = encodedFilename?.isEmpty != false ? "Unknown" : encodedFilename
```

Comment:
I think the previous logic you had might be a little clearer here since it might not be obvious to a reader that `nil != false`:
```
if let encodedFilename = aDecoder.decodeObject(...)?.lastPathComponent, !encodedFilename.isEmpty { ... }
```

Reply:
b24b0f235fc69751859bf1af6d73ff44ee443a6a